### DISCUSSION TITLE
Opentelemetry java agent disable without application restart

### DISCUSSION THREAD

**Discussion by ghost** [open-telemetry/opentelemetry-java-instrumentation#10950] [Q&A]
Hi , 

I would like to know all the possible options we can instrument otel java agent in microservices which could be managed at runtime without restart ,as we dont want any application downtime due to any challenges/error where we need to disable otel.

**Comment by steverao** [comment]
In OTel Java Agent, we control all abilities by configurations, you can find them in https://opentelemetry.io/docs/languages/java/automatic/configuration/ So far, we don't provide any approach to manage application without restart.

**Comment by trask** [comment]
hi @S484194, also check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9257

**Reply by ghost** [reply]
Is there any way i could write a premain or agentmain approach in my application which could be used to manage the agent dynamically. I believe this approach will have potential risks but would like to know any possible approaches if you guys are aware @steverao @trask
